keras/layers/core/permute.py

```diff
@@ -32,15 +32,14 @@ class Permute(Layer):
 
   ```python
   model = Sequential()
-  model.add(Permute((2, 1), input_shape=(10, 64)))
-  # now: model.output_shape == (None, 64, 10)
-  # note: `None` is the batch dimension
+  model.add(Permute((1, 3, 2), input_shape=(30, 10, 64)))
+  # now: model.output_shape == (None, 30, 64, 10)
+  # note: `30` is the batch dimension
```

Comment:
I don't think this is correct, the batch dim is not included in the input_shape, and it is still none in this case.

Reply:
Hey, as per the issue #41123, if the permute layer is added between the layers of the model such that we have a batch size, soo how shall we change the the dimensions of the input keeping the batch_size same. It is suggested in the issue to have a 3-D tensor example instead of a 2-D one because if we want to change the dimensions of the input keeping the batch_size same how can we do that?? Please correct me if I'm wrong. Thanks

Reply:
Note that the example you put here is a 4D tensor input (None, 30, 10, 64) where the batch dim size is None. When you mutate the shape, the output shape becomes (None, 30, 64, 10), and the batch dim is never changed here.